services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -53,104 +54,135 @@
-                project = predefined
+                new_project = predefined
 
         # re-validate data
-        projects_api.validate_project(request.app, project)
+        await projects_api.validate_project(request.app, new_project)
 
         # update metadata (uuid, timestamps, ownership) and save
-        project = await db.add_project(
-            project,
+        new_project = await db.add_project(
+            new_project,
             user_id,
             force_as_template=as_template is not None,
             hidden=hidden,
         )
 
         # copies the project's DATA IF cloned
         if clone_data_coro:
-            await clone_data_coro
+            assert source_project  # nosec
+            if as_template:
+                # we need to lock the original study while copying the data
+                async with projects_api.lock_with_notification(
+                    request.app,
+                    source_project["uuid"],
+                    ProjectStatus.CLONING,
+                    user_id,
+                    await get_user_name(request.app, user_id),
+                ):
+
+                    await clone_data_coro
+            else:
+                await clone_data_coro
+            # unhide the project if needed since it is now complete
+            if not new_project_was_hidden_before_data_was_copied:
+                await db.update_project_without_checking_permissions(
+                    new_project, new_project["uuid"], hidden=False
+                )
 
         # This is a new project and every new graph needs to be reflected in the pipeline tables
         await director_v2_api.create_or_update_pipeline(
-            request.app, user_id, project["uuid"]
+            request.app, user_id, new_project["uuid"]
         )
 
         # Appends state
-        project = await projects_api.add_project_states_for_user(
+        new_project = await projects_api.add_project_states_for_user(
             user_id=user_id,
-            project=project,
+            project=new_project,
             is_template=as_template is not None,
             app=request.app,
         )
 
     except ValidationError as exc:
+        log.debug("project validation error")
         raise web.HTTPBadRequest(reason="Invalid project data") from exc
     except ProjectNotFoundError as exc:
+        log.debug("project not found error")
         raise web.HTTPNotFound(reason="Project not found") from exc
     except ProjectInvalidRightsError as exc:
+        log.debug("project invalid rights error")
         raise web.HTTPUnauthorized from exc
-
+    except asyncio.CancelledError:
+        log.warning("cancelled creation of project, cleaning up")
+        await projects_api.delete_project(request.app, new_project["uuid"], user_id)
+        raise
     else:
-        raise web.HTTPCreated(text=json.dumps(project), content_type="application/json")
+        log.debug("project created successfuly")
```

Comment:
TIP: debug messages should be content richer and not only plain text. Use the trick i mentioned above with f-strings to display the value of variables in the log e.g.
```python
log.debug("project %s for %s created successfuly with %s", 
   f"{project_id=}", 
   f"{user_id=}", 
   f"{as_template=}",
)
```

Reply:
question.. what if the project_id does not exist?